### ISSUE TITLE
v3.2.0 no longer working on chrome

### ISSUE THREAD

**Issue by opeologist** [reduxjs/redux-devtools#1731]
chrome automatically updated this extension to v3.2.0 and it no longer sees/connects to stores on my site like the previous version did. no code changes in user land have occurred.

are there additional steps to enable the extension's use post-manifest v3? that looks to be the only change in v3.2.0

any help would be appreciated, thanks!

**Comment by Methuselah96** [comment]
Sorry that the new version is not working! There shouldn't be any additional steps to get it working, I will work on adding some configurable logging so that we can see what's going on. If you need a fix sooner, you can download the `chrome.zip` from [one of the earlier versions](https://github.com/reduxjs/redux-devtools/releases/tag/remotedev-redux-devtools-extensions%403.1.10) and use manually install that for now.

**Comment by opeologist** [comment]
wound up having to revert to v3.1.11 to unblock me while this gets sorted. checked before reverting if 3.2.3 fixed the issue and it was still not working. let me know if you need me to test anything related to resolving this! thank you!

**Comment by Methuselah96** [comment]
Can you check if there are any console errors for either the "Redux" DevTools tab or the background service worker?

You can check the "Redux" DevTools tab by right-clicking in the panel when you have the tab open and going to "Inspect View."

You can check the background service worker by going to "Extensions," enabling "Developer Mode," and clicking on the "service worker" text that appears in the Redux DevTools extensions square.

**Comment by Oikio** [comment]
I have similar issue, no errors in consoles of service worker or redux devtools.

One thing though, devtools works if it's tab with Redux Devtools was open/focused in Chrome DevTools focus when page reloads, but will fail to connect to store if one opens after page was loaded.

**Comment by Gianarci** [comment]
Same problem here. No error in console, but the plugin doesn't detect any store.

**Comment by Gianarci** [comment]
Same problem here with the 3.2.4!

**Comment by shadow1349** [comment]
I am having a similar issue with version 3.2.4. When I open the devtools I just get this
![image](https://github.com/user-attachments/assets/8e45ae1d-735a-4ce6-a25a-8b10521e96a5)


I am on Chrome (desktop) Version 128.0.6613.120 (Official Build) (arm64) and an M3 Mac

**Comment by PhantomLinker** [comment]
> I am having a similar issue with version 3.2.4. When I open the devtools I just get this ![image](https://private-user-images.githubusercontent.com/9685464/366082326-8e45ae1d-735a-4ce6-a25a-8b10521e96a5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwNDc2NTAsIm5iZiI6MTcyNjA0NzM1MCwicGF0aCI6Ii85Njg1NDY0LzM2NjA4MjMyNi04ZTQ1YWUxZC03MzVhLTRjZTYtYTI1YS04YjEwNTIxZTk2YTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MTFUMDkzNTUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjdkZGY1NTU5ZjZlMzA2NWQ0ZmNiNDgyZWQ4MTExN2Y4NmY5ODg5ZTY0YTA5ZDFiZmFiZGZhM2Q0ZmZkY2JmYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.aHvB0NjupT6ZJBFtGRzs9DbMw7KPTTk1FaNAXJYquLo)
> 
> I am on Chrome (desktop) Version 128.0.6613.120 (Official Build) (arm64) and an M3 Mac

I have the same issue after installing some updates today. Also Version 3.2.4 of the Extension. Chrome Version 128.0.6613.121 (Offizieller Build) (arm64) on an M1 Mac. 

The project runs NGXS instead of vanilla Redux or NgRx.

I found a workaround: Right Click on the extension -> "Open in a window" which should still display the Error Message -> reload/refresh the site so that the stores will load again =)

**Comment by PhantomLinker** [comment]
> @PhantomLinker, that's interesting. I don't get the error when opening in a new window but a refresh does show the store. However, when I close the extension and re-open it it breaks again.

Yes. Closing and Opening the extension will display the Error again. It might be a Bug in the Initialzation process of the extension.

**Comment by Gianarci** [comment]
Same problem here!

**Comment by MrTob** [comment]
same problem. Chrome Version 128.0.6613.138

**Comment by Methuselah96** [comment]
I've released [a new version](https://github.com/reduxjs/redux-devtools/releases/tag/remotedev-redux-devtools-extension%403.2.5) that adds logging to the background service worker.

Can someone log at the background service worker console log by going to "Extensions," enabling "Developer Mode," and clicking on the "service worker" text that appears in the Redux DevTools extensions square?

Here's a sample of what it looks like for me:
![image](https://github.com/user-attachments/assets/21fd9870-f152-4554-a697-3198a658fcb9)

**Comment by Mathyn** [comment]
@Methuselah96 Not sure I did it right but this is what I got.

![DebugRedux](https://github.com/user-attachments/assets/af1c0b87-52c8-4a93-8ec8-7f40e0480a77)

As a side note: the plugin never works for me when I click the dedicated extension button (it always claims it cannot find a store).

The extension DOES work for me if I open my browser console (F12), then navigate to the redux tab and then refresh my browser page.

For the second case I am unsure how to get the debug logs, if you can help me with that I can provide these as well (perhaps if you compare the two that provides some useful info?).

**Comment by uksparky** [comment]
I've had similar issues as described above, when using autoselect instances and reloading a page I get the attached. 
![Screenshot 2024-09-17 at 12 12 07](https://github.com/user-attachments/assets/d3c1db78-2ba9-4e11-8bf9-b724032730fc)

**Comment by Methuselah96** [comment]
Thanks! I'm able to reproduce the issue with NGXS, looking into that.

**Comment by VitorHFLopes** [comment]
> @Methuselah96 Not sure I did it right but this is what I got. I am using this plugin in combination with NGXS.
> 
> ![DebugRedux](https://private-user-images.githubusercontent.com/4532865/368086215-af1c0b87-52c8-4a93-8ec8-7f40e0480a77.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY2MjcxMTEsIm5iZiI6MTcyNjYyNjgxMSwicGF0aCI6Ii80NTMyODY1LzM2ODA4NjIxNS1hZjFjMGI4Ny01MmM4LTRhOTMtOGVjOC03ZjQwZTA0ODBhNzcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MThUMDIzMzMxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTVhZTZiNGMwNWM4ZjNkOGZlNDU3NzcyZGQxYTM0ZjgwM2I4MzVmMjFjMzE3MWI1ODc0OTNjZWFmZjc0ODIzMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.mC7DKV5SdMXfwkB5MZuZulke0EEDFf8W1TsT0GHNMm4)
> 
> As a side note: the plugin never works for me when I click the dedicated extension button (it always claims it cannot find a store).
> 
> The extension DOES somewhat work for me if I open my browser console (F12), then navigate to the redux tab and then refresh my browser page.
> 
> With 'somewhat' I mean:
> 
> * It loads the current state of the stores
> * It does not show any new actions being dispatched
> * It does not update the state of the stores beyond the initial state
> 
> For the second case I am unsure how to get the debug logs, if you can help me with that I can provide these as well (perhaps if you compare the two that provides some useful info?).

For now, that saves a lot!

**Comment by Methuselah96** [comment]
I've looked into this, unfortunately I'm not sure there's a trivial fix. The issue is due to changes in Manifest V3, namely that background pages are no longer persistent, so we can no longer store past state and actions on the background page.

This isn't a problem with normal Redux because we're already storing the past state and actions on the page, so when the Redux DevTools window is opened, it just requests all that past data. It doesn't look like the NGXS DevTools plugin is currently set up to do this. It worked with Manifest V2 because all past state and actions were stored on the background page, so it would just use that whenever the Redux DevTools window was opened.

I'll have to give this one some thought. Thank you all for your patience.

**Comment by Methuselah96** [comment]
Is there anyone here who is not using NGXS, or another custom integration? Just checking to make sure that's the main issue here.

**Comment by opeologist** [comment]
> Is there anyone here who is not using NGXS, or another custom Redux DevTools integration? Just checking to make sure that's the main issue here.

we're using zustand if it helps

**Comment by razvanilin** [comment]
> Is there anyone here who is not using NGXS, or another custom Redux DevTools integration? Just checking to make sure that's the main issue here.

I'm having similar issues in a Vite + React + Redux Toolkit setup. The store gets connected 90% of the time, but the main issue is that the Redux DevTools tab stops updating with new dispatches unless I close the devtools and open it again

**Comment by uksparky** [comment]
> Is there anyone here who is not using NGXS, or another custom Redux DevTools integration? Just checking to make sure that's the main issue here.

I am and the console image i previously attached is from a plain redux app with redux devtools extension. 

My symptoms are similar to those of @razvanilin 

It seems like some sort of connection is timing out after a while as i can be fairly confident if I have devtools open and manage to connect it to an instance then after a few minutes that instance no longer sees any dispatched actions. If I then close dev tools and reopen then I get the white loading circle.

**Comment by Methuselah96** [comment]
Thanks all! Sounds like we have two main issues:

- Some third-party integrations (e.g., NGXS and Zustand) no longer work
  - Current workaround is to have the Redux DevTools tab open before loading the web page and/or refresh your web page once the tab is open
- DevTools loses connection if it remains inactive for too long
  - Current workaround is to refresh the web page, but that's obviously not ideal

Let me know if you have an issue that doesn't seem like it fits in either category.

I'll be working hard on fixes and post any updates here, sorry for the interruption. Both of this issues are not trivial and are unfortunate by-products of the loss of a persistent background page in Manifest V3.

**Comment by zehawki** [comment]
@Methuselah96 thank you for looking into these issues!

I'm having a similar (but different) issue on 
1. Chrome 128.0.6613.138 and Brave [Version 1.69.168 Chromium: 128.0.6613.138]
2. Redux DevTools 3.2.5
3. Vanilla react setup: React 17 + CRA

Redux DevTools just freezes completely upon hot reloading. After that irrespective of how many times I reload, nothing will refresh. The only way to recover is to close dev tools and reopen. Once I reopen dev tools, Redux DevTools connects just fine to the store, and shows me the existing (past) actions, states, etc etc and works fine after that. Until the next hot reload. At which point I have to do this rigmarole again.

**Comment by aghiuta-gaggle** [comment]
> I've looked into this, unfortunately I'm not sure there's a trivial fix. The issue is due to changes in Manifest V3, namely that background pages are no longer persistent, so we can no longer store past state and actions on the background page.
> 
> This isn't a problem with normal Redux because we're already storing the past state and actions on the page, so when the Redux DevTools window is opened, it just requests all that past data. It doesn't look like the NGXS DevTools plugin is currently set up to do this. It worked with Manifest V2 because all past state and actions were stored on the background page, so it would just use that whenever the Redux DevTools window was opened.
> 
> I'll have to give this one some thought. Thank you all for your patience.

@Methuselah96 I've encountered the same issue with Manifest V3 and here's something that helped keeping it alive:

```
// https://stackoverflow.com/questions/66618136/persistent-service-worker-in-chrome-extension
// https://bugs.chromium.org/p/chromium/issues/detail?id=1152255
const keepAlive = () => setInterval(chrome.runtime.getPlatformInfo, 20e3);
chrome.runtime.onStartup.addListener(keepAlive);
keepAlive();
```

**Comment by andyghiuta** [comment]
@Methuselah96 I've encountered the same issue with Manifest V3 and here's something that helped keeping it alive:

```
// https://stackoverflow.com/questions/66618136/persistent-service-worker-in-chrome-extension
// https://bugs.chromium.org/p/chromium/issues/detail?id=1152255
const keepAlive = () => setInterval(chrome.runtime.getPlatformInfo, 20e3);
chrome.runtime.onStartup.addListener(keepAlive);
keepAlive();
```

**Comment by MaksimAliakhnovich** [comment]
1. Chrome 128.0.6613.138 
2. Redux DevTools 3.2.5
3. NextJs 14+

![image](https://github.com/user-attachments/assets/77604593-ff25-4de6-86a0-ea8ee96273c9)
this screenshot is after disabling, the development panel was closed. At some point it turns off and does not respond to buttons, events do not come, you can not make a commit to clear history.

but when the developer panel was opened everything worked fine even after a long time. as it seemed to me

maybe this will help with something

**Comment by Methuselah96** [comment]
The DevTools losing connection should hopefully be fixed in [v3.2.6](https://github.com/reduxjs/redux-devtools/releases/tag/remotedev-redux-devtools-extension%403.2.6), let me know anyone still has issues with that

**Comment by uksparky** [comment]
> The DevTools losing connection should hopefully be fixed in [v3.2.6](https://github.com/reduxjs/redux-devtools/releases/tag/remotedev-redux-devtools-extension%403.2.6), let me know if you are still experiencing that behavior

I used it briefly this afternoon for a few hours and it seemed to be working well, good work dude.

**Comment by Mathyn** [comment]
> The DevTools losing connection should hopefully be fixed in [v3.2.6](https://github.com/reduxjs/redux-devtools/releases/tag/remotedev-redux-devtools-extension%403.2.6), let me know if you are still experiencing that behavior

Just gave it a quick try and the issues seem to be fixed when using NGXS. Will continue to test the rest of the day but so far looks really good. Nice job man!

**Comment by opeologist** [comment]
ty for the updates! i'm able to see my store again on the latest version of the extension without issue. i think this can be closed as successful. thanks again!

**Comment by shadow1349** [comment]
Hey there, thanks for all the effort! I'm trying on version 3.2.6 and it seems to only work when I open it in a new tab and not in dev tools or when I use the pop up.

![image](https://github.com/user-attachments/assets/52b3eb35-a1d6-4f55-a065-19d91fd84877)
![image](https://github.com/user-attachments/assets/15b328cc-1910-4be3-b013-185015f927ba)

I'm not sure how to get logs for those. Let me know how I can provide any additional details.

**Comment by Methuselah96** [comment]
@shadow1349 Are you using Redux, or a different state management library? Can you try v3.2.7 (by removing and re-adding the extension from the Chrome Web Store)?

**Comment by shadow1349** [comment]
Just updated, 3.2.7 works perfectly. Thank you!

**Comment by Methuselah96** [comment]
I've finished rolling out v3.2.7 to all users, so I believe the main issues here should be fixed

**Comment by Methuselah96** [comment]
@zehawki I tried reproducing your issue with hot reloading and CRA, but was unsuccessful. Can you try out the new version, and if you're still experiencing problems, can you create a new issue?

**Comment by markerikson** [comment]
@nhhockeyplayer Honestly that comment is pretty incoherent and not at all relevant to this issue :(

If you think there is a specific problem, please file a new issue a meaningful explanation and with steps to reproduce.

We haven't changed any of the global methods that libraries like NgRx would use to display their values, nor any of the state handling.
